policy/modules/services/pacemaker.if

```diff
@@ -1,5 +1,25 @@
 ## <summary>A scalable high-availability cluster resource manager.</summary>
 
+########################################
+## <summary>
+##   Make the specified unit type contronable
+##   by pacemaker.
+## </summary>
+## <param name="type">
+##   <summary>
+##   Type to be used for pacemaker controllable unit.
+##   </summary>
+## </param>
+#
+interface(`pacemaker_unit_file',`
+	gen_require(`
+		attribute pacemakerunit;
+	')
+
+	init_unit_file($1)
+	typeattribute $1 pacemakerunit;
+')
```

Comment:
With no callers of this in the PR, I'm not convinced this is the right way to go.

Reply:
Ok, how else should this be done?  I used it in a module within a policy module specific to the project being worked on.  Pacemaker starts/stops a service as the particular node in the cluster becomes primary.  

Rather than an attribute, this could be an interface like 'pacemaker_startstop_unit' (or similar) and the rule in the interface could be 'allow pacemaker_t $1:service { start stop };'    

I don't see any existing interfaces that have this model.  The existing interfaces that are similar are all the reverse where some type is granted permission to start/stop a known service (i.e. clamav_startstop_clamd, ntp_startstop, ntp_startstop, etc..)

It seems too heavy handed to use init_start_all_units(pacemaker_t) and init_stop_all_units(pacemaker_t), but that would also work.

Thoughts?

Reply:
With no example, I can't tell if an alternate concept for this would work. I don't want to add something while potentially having to deprecate it as soon as a real example comes along.

Reply:
Ok, I will change this portion of the patch to model it after how the monit module does something similar.  In the monit case there is a tunable (defaulted to false) that when enabled allows monit to start/stop any service.  I will update and we can then decide if that is OK or if a different approach is preferable.

Reply:
I think we should probably have and keep this tunable even if we add something like the `packemaker_unit_file`.

Reply:
I have updated this with the tunable.  I agree, it would stay even if we do get to an interface for this at some point.